vagrant/provisioning/roles/haproxy/tasks/main.yml

```diff
@@ -226,6 +229,107 @@
+        use_backend be_elasticsearch if acl_elasticsearch
+        use_backend be_gateway if acl_gateway
+        use_backend be_eureka if acl_eureka
+        use_backend be_eureka-dashboard if acl_eureka-dashboard
+        use_backend be_control-center if acl_control-center
+        use_backend be_zipkin-health if acl_zipkin-health
+        use_backend be_zipkin-api if acl_zipkin-api
+        use_backend be_zipkin if acl_zipkin
+        use_backend be_config if acl_config
+    - name: Populate haproxy.conf with backend servers
+      become: yes
+      marker: "# {mark} ANSIBLE MANAGED BLOCK MICROSERVICES CONTEXT"
+      insertafter: "#backend servers"
+      path: "/etc/opt/rh/rh-haproxy18/haproxy/haproxy.cfg"
+      block: |
+        backend be_schema-registry
+          http-response add-header X-Schema-registry %s
+          mode http
+          option tcp-check
+          server schema-registry {{ internal_host }}:8081
+          reqrep ^([^\ ]*\ /)schema-registry[/]?(.*)     \1\2
+
+        backend be_elasticsearch
+          http-response add-header X-Elasticsearch %s
+          mode http
+          option tcp-check
+          server elasticsearch {{ internal_host }}:9200
+
+        backend be_gateway
+          http-response add-header X-Gateway %s
+          mode http
+          option tcp-check
+          server gateway {{ arkcase_host }}:9000
+
+        backend be_eureka
+          http-response add-header X-Eureka %s
+          mode http
+          option tcp-check
+          server eureka {{ arkcase_host }}:9000/eureka
+
+        backend be_eureka-dashboard
+          http-response add-header X-Eureka-Dashboard %s
+          mode http
+          option tcp-check
+          server eureka-dashboard {{ arkcase_host }}:9000/eureka-dashboard
+
+        backend be_control-center
+          http-response add-header X-Control-Center %s
+          mode http
+          option tcp-check
+          server control-center {{ internal_host }}:488 check fall 3 rise 2 check ssl verify required ca-file {{ ssl_ca }}
+
+        backend be_zipkin-health
+          http-response add-header X-Zipkin-Health %s
+          mode http
+          option tcp-check
+          server zipkin-health {{ internal_host }}:9411/health
+
+        backend be_zipkin-api
+          http-response add-header X-Zipkin-Api %s
+          mode http
+          option tcp-check
+          server zipkin-api {{ internal_host }}:9411/api
+
+        backend be_zipkin
+          http-response add-header X-Zipkin %s
+          mode http
+          option tcp-check
+          server zipkin {{ internal_host }}:9411/zipkin
+
+        backend be_config
+          http-response add-header X-Config %s
+          mode http
+          option tcp-check
+          server config {{ arkcase_host }}:9999
+
+  when: enable_kafka is defined and enable_kafka == "yes" and enable_elasticsearch is defined and enable_elasticsearch == "yes"        
```

Comment:
It sounds like we need to separate the kafka lines and the elasticsearch lines .... if enable_kafka is "yes" and enable_elasticsearch is "no" then we don't get the Kafka lines, and if enable_kafka is "no" and enable_elasticsearch is "yes" then we don't get the elastic lines.

Reply:
I will separate kafka lines from elasticsearch lines and vice versa